### ISSUE TITLE
Add timezone to log file output or switch to UTC

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/rethinkdb#4950]
The timestamps in the log file should have the timezone printed with each timestamp. This should probably be the current offset from UTC, so we can detect whether an entry is in daylight saving time or not.

We might be able to use ISO8601 timestamps, possibly slightly modified / enhanced to have a higher sub-second precision for our timestamps (since we require them to be unique).

Ideally we'll do this in a way where our parser can distinguish between the old and new format, so that old log files can still be parsed.

**Comment by danielmewes** [comment]
We can alternatively switch our current time stamp format to using UTC timestamps, but that's harder to interpret for humans.

**Comment by encryptio** [comment]
Pedantically, [RFC 3339](https://www.ietf.org/rfc/rfc3339.txt) is a strict subset of ISO8601 which only supports the one format we'd like to write them in (e.g. no "iso week" madness.) It (and ISO8601, of course) supports arbitrary precision subsecond fields.

**Comment by larkost** [comment]
I am very much in favor of ISO8601 (or RFC 3339), done to the systems local time since it is both easier on the users, and less ambiguous for all readers.

**Comment by mlucy** [comment]
I agree with ISO 8601 timestamps (and we already have code to produce them for the related ReQL terms), but I think we should consider always outputting UTC timestamps.

Putting UTC timestamps in our logs makes it easier to compare logs on servers in different geographic regions, and it also preserves the nice property that log entries are in lexicographic order.

**Comment by larkost** [comment]
I disagree on the UTC output to files. While it makes it a bit easier for someone who is comparing the raw files from multiple timezones, those people are the ones most likely to be using logging tools to aggregate there logs anyways, obliviating any advantage. For those with smaller clusters all in a single timezone (usually the one they are sitting in) having it in local time is much easier.

For our combined view in `rethinkdb.logs` I agree that we should display a single timezone, ideally the one of the user looking at the logs. But absent a good way to determine the proper timezone, I would be fine with UTC timezone in the outputted logs.

**Comment by danielmewes** [comment]
It looks like our `logs` system table already uses UTC as the default timezone.

@nighelles is currently working on the log system. We should use this as a chance to either switch to RFC 3339 dates with a timezone specification, or to UTC timestamps in the log file. I don't particularly care which, and users will be able to customize their output through syslog if it's important for them anyway.
When changing the format, we should make sure that we don't fail when reading old log files. through the `logs` table. I think seeing wrong timezones for the old log entries (or even not seeing them at all) is acceptable though.
